lib/smack/MemoryAllocationChecker.cpp

```diff
@@ -0,0 +1,54 @@
+//
+// This file is distributed under the MIT License. See LICENSE for details.
+//
+#include "smack/MemoryAllocationChecker.h"
+#include "smack/Naming.h"
+
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/Support/Debug.h"
+
+namespace smack {
+  using namespace llvm;
+
+  const char* VERIFIER_FUNCTION = "__SMACK_check_memory_access";
+
+  bool MemoryAllocationChecker::runOnModule(Module& m) {
+    DataLayout* dataLayout = new DataLayout(&m);
+    Function* verifierFunction;
+    for(auto& f:m) {
+      if(f.getName() == VERIFIER_FUNCTION) {
+        verifierFunction = &f;
+        break;
+      }
+    }
+    for (auto& F : m) {
+      if(!Naming::isSmackName(F.getName())) {
+        for (auto& B : F) {
```

Comment:
Could you get rid of the loop that goes over basic blocks by iterating directly over instructions?
See: http://llvm.org/docs/ProgrammersManual.html#iterating-over-the-instruction-in-a-function